nixos/modules/services/networking/firewall.nix

```diff
@@ -493,6 +511,80 @@ in
         '';
     };
 
+    networking.firewall.extraAllowed = mkOption {
+      type = types.listOf (types.submodule { options = {
+        sourceAddress = mkOption {
+          type = types.nullOr types.string;
```

Comment:
Is the choice of `types.string` deliberate? Usually `str` is preferred for plain strings.